src/java/org/apache/cassandra/io/sstable/format/SSTableReader.java

```diff
@@ -626,42 +575,25 @@ public void run()
      * Open a RowIndexedReader which already has its state initialized (by SSTableWriter).
      */
     public static SSTableReader internalOpen(Descriptor desc,
-                                      Set<Component> components,
-                                      TableMetadataRef metadata,
-                                      FileHandle ifile,
-                                      FileHandle dfile,
-                                      IndexSummary isummary,
-                                      IFilter bf,
-                                      long maxDataAge,
-                                      StatsMetadata sstableMetadata,
-                                      OpenReason openReason,
-                                      SerializationHeader header)
-    {
-        assert desc != null && ifile != null && dfile != null && isummary != null && bf != null && sstableMetadata != null;
-
-        SSTableReader reader = internalOpen(desc, components, metadata, maxDataAge, sstableMetadata, openReason, header);
-
-        reader.bf = bf;
-        reader.ifile = ifile;
-        reader.dfile = dfile;
-        reader.indexSummary = isummary;
-        reader.setup(true);
-
-        return reader;
-    }
-
-
-    private static SSTableReader internalOpen(final Descriptor descriptor,
-                                              Set<Component> components,
-                                              TableMetadataRef metadata,
-                                              Long maxDataAge,
-                                              StatsMetadata sstableMetadata,
-                                              OpenReason openReason,
-                                              SerializationHeader header)
-    {
-        Factory readerFactory = descriptor.getFormat().getReaderFactory();
-
-        return readerFactory.open(descriptor, components, metadata, maxDataAge, sstableMetadata, openReason, header);
+                                             Set<Component> components,
+                                             TableMetadataRef metadata,
+                                             FileHandle ifile,
+                                             FileHandle dfile,
+                                             IFilter bf,
+                                             long maxDataAge,
+                                             StatsMetadata sstableMetadata,
+                                             OpenReason openReason,
+                                             SerializationHeader header,
+                                             IndexSummary summary)
+    {
+        assert desc != null && ifile != null && dfile != null && summary != null && bf != null && sstableMetadata != null;
+
+        BuilderForWriter builder = new BuilderForWriter(desc, metadata, maxDataAge);
+        builder.bf = bf;
+        builder.ifile = ifile;
+        builder.dfile = dfile;
+        builder.summary = summary;
```

Comment:
nit: We could make a minor clean up here and have "chainable" methods for setting the bloom filter if you think that reads better.

ex.
```
BuilderForWriter builder =
    new BuilderForWriter(desc, metadata, maxDataAge).bloomFilter(bf)
                                                    .indexFile(ifile)
                                                    .dataFile(dfile)
                                                    .indexSummary(summary);
```

Reply:
good idea.